### PR TITLE
[TextFields] Add placeholderLabel accessibilityLabel to textField accessibilityValue

### LINKED ISSUE #4315: [Chips] TextField chips should include placeholder in VoiceOver readout on selection
Similarly to textfields, the placeholder should be read out by voice over on selection.

### PR DESCRIPTION
**Pull request by andrewoverton**
It's a `Chips` issue, but since `MDCChipField` uses an `MDCTextField` I think adding the functionality described in the issue to the textfield will address it.

Will resolve https://github.com/material-components/material-components-ios/issues/4315

### CODE REVIEW

**Comment by jverkoey** [comment]
Can we delineate the different strings in some manner so that when they're read aloud it doesn't sound like one long sentence?

File: components/TextFields/src/MDCTextField.m [hunk 1]
```diff
@@ -702,12 +702,18 @@ - (void)mdc_setAdjustsFontForContentSizeCategory:(BOOL)adjusts {
 }
 
 - (NSString *)accessibilityValue {
-  if (self.leadingUnderlineLabel.text.length > 0) {
-    return [NSString stringWithFormat:@"%@ %@", [super accessibilityValue],
-                                      self.leadingUnderlineLabel.accessibilityLabel];
+  NSMutableArray *accessibilityStrings = [NSMutableArray new];
```
Comment (ianegordon) [inline components/TextFields/src/MDCTextField.m]: We tend to use alloc-init over new.

Comment (romoore) [reply components/TextFields/src/MDCTextField.m]: Or `[NSMutableArray array]` for arrays.

**Review by romoore — APPROVED** [review]
Please add unit tests to cover the `accessibilityValue` method.

**Comment by andrewoverton** [comment]
@romoore  and I observed some strange behavior with the MDCTextField's placeholder UILabel's accessibilityValue that got in the way of unit testing. For some reason the placeholder label's accessibilityValue was nil even though the text was set. This was inconsistent with UILabel behavior elsewhere, including the Catalog example.

**Pull request merged**